custodia/secrets.py

```diff
@@ -257,7 +257,12 @@ def _int_get_key(self, trail, request, response):
         key = self._db_key(trail)
         try:
             output = self.root.store.get(key)
-            if output is None:
+            # NOTE(rmascena): If the response value is empty, the
+            # row is a subcontainer, so only returns if it is
+            # requested properly.
+            if not output or output is None:
+                self.logger.error('GET a container should terminate '
+                                  'with a "/" in the request. ')
                 raise HTTPError(404)
```

Comment:
So now looking at the error code ... 404 is not really correct if you want to return an error.
I think we may have to split the test. See API.md for details on what error codes mean.
if None is returned then we return 404, but if the query is not acceptable we should probably return 406.
So the pseudo-code would be:
if output is None:
    return 404
if len(output) == 0:
    <log>
    return 406

or something along thse lines.